polkadot/zombienet-sdk-tests/build.rs

```diff
@@ -0,0 +1,151 @@
+	let manifest_path = manifest_path.strip_suffix('/').unwrap();
+	debug_output!("manifest_path is  : {}", manifest_path);
+	let package = format!("{chain}-runtime");
+	let profile = PROFILES.into_iter().find(|p| {
+		let full_path = format!(
+			"{}/target/{}/wbuild/{}/{}.wasm",
+			manifest_path,
+			p,
+			&package,
+			replace_dashes(&package)
+		);
+		debug_output!("checking wasm at : {}", full_path);
+		matches!(path::PathBuf::from(&full_path).try_exists(), Ok(true))
+	});
+
+	debug_output!("profile is : {:?}", profile);
+	profile.map(|profile| {
+		PathBuf::from(&format!(
+			"{}/target/{}/wbuild/{}/{}.wasm",
+			manifest_path,
+			profile,
+			&package,
+			replace_dashes(&package)
+		))
+	})
+}
+
+// based on https://gist.github.com/s0me0ne-unkn0wn/bbd83fe32ce10327086adbf13e750eec
+fn build_wasm(chain: &str) -> PathBuf {
+	let package = format!("{chain}-runtime");
+
+	let cargo = env::var("CARGO").unwrap();
+	let target = env::var("TARGET").unwrap();
+	let out_dir = env::var("OUT_DIR").unwrap();
+	let target_dir = format!("{}/runtimes", out_dir);
+	let args = vec![
+		"build",
+		"-p",
+		&package,
+		"--profile",
+		"release",
+		"--target",
+		&target,
+		"--target-dir",
+		&target_dir,
+	];
+	debug_output!("building metadata with args: {}", args.join(" "));
+	Command::new(cargo)
+		.env_remove("SKIP_WASM_BUILD") // force build to get the metadata
+		.args(&args)
+		.status()
+		.unwrap();
+
+	let wasm_path = &format!(
+		"{target_dir}/{target}/release/wbuild/{}/{}.wasm",
+		&package,
+		replace_dashes(&package)
+	);
+	PathBuf::from(wasm_path)
+}
+
+fn generate_metadata_file(wasm_path: &Path, output_path: &Path) {
+	let source = Source::from_options(Some(wasm_path.to_path_buf()), None, None, None).unwrap();
+	let subwasm = Subwasm::new(&source.try_into().unwrap()).unwrap();
+	let mut output_file = std::fs::File::create(output_path).unwrap();
+	subwasm.write_metadata(OutputFormat::Scale, None, &mut output_file).unwrap();
+}
+
+fn fetch_metadata_file(chain: &str, output_path: &Path) {
+	// First check if we have an explicit path to use
+	let env_key = format!("{}_METADATA_FILE", make_env_key(chain));
+
+	if let Ok(path_to_use) = env::var(env_key) {
+		debug_output!("metadata file to use (from env): {}\n", path_to_use);
+		let metadata_file = PathBuf::from(&path_to_use);
+		fs::copy(metadata_file, output_path).unwrap();
+	} else if let Some(exisiting_wasm) = find_wasm(chain) {
+		debug_output!("exisiting wasm: {:?}", exisiting_wasm);
+		// generate metadata
+		generate_metadata_file(&exisiting_wasm, output_path);
+	} else {
+		// build runtime
+		let wasm_path = build_wasm(chain);
+		debug_output!("created wasm: {:?}", wasm_path);
+		// genetate metadata
+		generate_metadata_file(&wasm_path, output_path);
+	}
+}
+
+fn main() {
+	if env::var("CARGO_FEATURE_ZOMBIE_METADATA").is_err() {
+		debug_output!("zombie-metadata feature not enabled, not need to check metadata files.");
```

Comment:
not needed?

Reply:
Since this test needs extra artifacts (e.g. metadata files), we will only run if the feature `zombie-metadata` is enabled to reduce the overhead outside our CI.